src/user-guide/writing-workflows/scheduling.rst

```diff
@@ -1966,189 +1714,86 @@ When using message triggers in this way there are two things to be aware of:
+the upstream tasks:
 
 .. code-block:: cylc
 
    [scheduling]
+       initial cycle point = 2025-01-01T00
        [[graph]]
-           T00,T06,T12,T18 = "A[-PT6H] => A"
-
-If your model is configured to write out additional restart files
-to allow one or more cycle points to be skipped in an emergency *do not
-represent these potential dependencies in the workflow graph* as they
-should not be used under normal circumstances. For example, the
-following graph would result in task ``A`` erroneously
-triggering off ``A[T-24]`` as a matter of course, instead of
-off ``A[T-6]``, because ``A[T-24]`` will always
-be finished first:
-
-.. code-block:: cylc
-
-   [scheduling]
-       [[graph]]
-           # DO NOT DO THIS (SEE ACCOMPANYING TEXT):
-           T00,T06,T12,T18 = "A[-PT24H] | A[-PT18H] | A[-PT12H] | A[-PT6H] => A"
-
-
-How The Graph Determines Task Instantiation
--------------------------------------------
-
-A graph trigger pair like ``foo => bar`` determines the existence and
-prerequisites (dependencies) of the downstream task ``bar``, for
-the cycle points defined by the associated graph section heading. In general it
-does not say anything about the dependencies or existence of the upstream task
-``foo``. However *if the trigger has no cycle point offset* Cylc
-will infer that ``bar`` must exist at the same cycle points as
-``foo``. This is a convenience to allow this:
-
-.. code-block:: cylc
-
-   R1 = "foo => bar"
-
-to be written as shorthand for this:
-
-.. code-block:: cylc
-
-   R1 = """
-       foo
-       foo => bar
-   """
-
-(where ``foo`` by itself means ``<nothing> => foo``, i.e. the
-task exists at these cycle points but has no prerequisites - although other
-prerequisites may be defined for it in other parts of the graph).
-
-*Cylc does not infer the existence of the upstream task in offset
-triggers* like ``foo[-P1D] => bar`` because a typo in the offset
-interval should generate an error rather than silently creating
-tasks on an erroneous cycling sequence.
+           P2D = "foo & bar => baz"
+  
+This says ``baz`` depends on ``foo`` and ``bar`` for every point in the
+sequence defined by the recurrence ``P2D`` (i.e. ``R/^/P2D``).
 
-As a result you need to be careful not to define inter-cycle dependencies that
-cannot be satisfied at run time. Workflow validation catches this kind of error if
-the existence of the cycle offset task is not defined anywhere at all:
+Cycle does not infer the cycling sequence for upstream tasks in intercycle triggers,
+however. This prevents mistakes in the graph (e.g. ``P1D`` should be ``P2D`` above) from
+silently creating tasks on off-sequence points:
 
 .. code-block:: cylc
 
    [scheduling]
-       initial cycle point = 2020
+       initial cycle point = 2025-01-01T00
        [[graph]]
-           # ERROR
-           P1Y = "foo[-P1Y] => bar"
-
-.. code-block:: console
+           # ERROR: No cycling sequences defined for foo
+           P2D = "foo[-P1D] & bar
+ 
+.. warning::
```

Comment:
This doesn't make much sense to me
- Why does it say
  > ``P1D`` should be ``P2D`` above

  (there is no `P1D` above)

- Why is it talking about
  > creating tasks on off-sequence points
  
  when the error would be the same if it was `P2D = foo [-P2D] & bar`, right?

Reply:
Sorry, that was a bit of a mess. Try it for size now.